migration-0.38/migration/alerts.py

```diff
@@ -0,0 +1,105 @@
+import sqlite3
+import json
+from clickhouse_driver import Client
+import sys
+import json
+from migration.fields import update_field
+
+
+
+def update_alert(data, fields):
+
+    query_type = data['condition']['compositeQuery']['queryType']
+    if query_type == "clickhouse_sql":
+        # print(testing)
+        # there will only be one query for ch in alerts
+        clickhouse_sql = data['condition']['compositeQuery']['chQueries']['A']
+        if "query" not in clickhouse_sql.keys():
+            print("Query not found : {} , type: {}".format(data['alert'], query_type))
+            print(clickhouse_sql)
+            return data
+        clickhouse_sql =clickhouse_sql["query"]
+        
+        if "signoz_logs.distributed_logs"  in clickhouse_sql:
+            print("Alert : {} , type: {}".format(data['alert'], query_type))
+            for old_name, updated_attribute in fields.items():
+                # check if attribute is there
+                # check is done by checking # attributes_string_key, 'request_context_test_name'
+                check = updated_attribute[2] + "_" + updated_attribute[1].lower() + "_key, '" + updated_attribute[0].replace('.', '_') + "'"
+                if check in clickhouse_sql:
+                    updated = updated_attribute[2] + "_" + updated_attribute[1].lower() + "_key, '" + updated_attribute[0] + "'"
+                    clickhouse_sql = clickhouse_sql.replace(check, updated)
+
+                
+                # check if materialized column is there if yes then replace it
+                materialized_name = updated_attribute[2][:-1] + "_" + updated_attribute[1].lower() + "_" + updated_attribute[0].replace('.', '_')
```

Comment:
Just for clarity, `updated_attribute[2][:-1]` for `attributes` -> `attribute`?

Reply:
correct